instance/migrations/0043_change_server_statuses.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('instance', '0042_add_instance_status'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='openstackserver',
+            name='_status',
+            field=models.CharField(choices=[('booting', 'Booting'), ('building', 'Building'), ('failed', 'BuildFailed'), ('pending', 'Pending'), ('ready', 'Ready'), ('terminated', 'Terminated'), ('unknown', 'Unknown')], default='pending', db_column='status', max_length=20, db_index=True),
+        ),
+        migrations.RunSQL(
+            [
+                "UPDATE instance_openstackserver SET status = 'pending' WHERE status = 'new'",
+                "UPDATE instance_openstackserver SET status = 'building' WHERE status = 'started'",
+                "UPDATE instance_openstackserver SET status = 'booting' WHERE status = 'active' OR status = 'rebooting'",
+                "UPDATE instance_openstackserver SET status = 'ready' WHERE status = 'booted' OR status = 'provisioning'",
```

Comment:
@bradenmacdonald Thanks for pointing this out. I think we mainly need to attack the issue of incorporating `progress` info in the migration that precedes this one (i.e., `0042_add_instance_status.py`): Servers don't track the provisioning process anymore, so it is fine for a server to have status `Ready` even if the provisioning process failed. And `failed` (`BuildFailed`) is supposed to indicate that OpenStack failed to create the server, so it wouldn't be correct to set "Provisioning Failed" to `failed`. So we need to use appropriate instance states (`Error` and `ConfigurationFailed`) to indicate build failures and provisioning failures. This migration should only take into account `progress` info for servers with a status of `started`; if "Started Failed", status needs to be set to `failed`. (I couldn't find any transitions from "Started Running" to "Started Failed" on master, so modifying this migration to that extent would mostly be a safety net.)